app/Controller/OAuthController.php

```diff
@@ -106,7 +106,13 @@ public function unlink()
     protected function authenticate($providerName)
     {
         if ($this->authenticationManager->oauthAuthentication($providerName)) {
-            $this->response->redirect($this->helper->url->to('DashboardController', 'show'));
+            if (isset($this->sessionStorage->redirectAfterLogin) && ! empty($this->sessionStorage->redirectAfterLogin) && ! filter_var($this->sessionStorage->redirectAfterLogin, FILTER_VALIDATE_URL)) {
```

Comment:
The class `SessionStorage` doesn't exists since Kanboard 1.2.

You have a starting point here to add this feature: https://github.com/kanboard/kanboard/blob/master/app/Controller/AuthController.php#L73-L83